twisted/test/test_iutils.py

```diff
@@ -226,7 +227,7 @@ def _defaultPathTest(self, utilFunc, check):
         os.chmod(dir, 0)
 
         d = utilFunc(self.exe, ['-u', scriptFile])
-        d.addCallback(check, dir)
+        d.addCallback(check, dir.encode())
```

Comment:
I'm nervous about these two naked `encode` calls as well. Is there a sensible encoding we can choose here? Maybe `sys.getfilesystemencoding()`?